src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/DoNotUseInsecureDeserializerMethodsBase.cs

```diff
@@ -70,12 +88,17 @@ public sealed override void Initialize(AnalysisContext context)
                             if (invocationOperation.Instance?.Type?.DerivesFrom(deserializerTypeSymbol) == true
                                 && cachedDeserializationMethodNames.Contains(invocationOperation.TargetMethod.MetadataName))
                             {
-                                operationAnalysisContext.ReportDiagnostic(
-                                    Diagnostic.Create(
-                                        this.MethodUsedDescriptor,
-                                        invocationOperation.Syntax.GetLocation(),
-                                        invocationOperation.TargetMethod.ToDisplayString(
-                                            SymbolDisplayFormat.MinimallyQualifiedFormat)));
+                                DiagnosticDescriptor? chosenDiagnostic =
+                                    this.ChooseDiagnosticDescriptor(operationAnalysisContext, wellKnownTypeProvider);
+                                if (chosenDiagnostic != null)
+                                {
+                                    operationAnalysisContext.ReportDiagnostic(
+                                        Diagnostic.Create(
+                                            chosenDiagnostic,
+                                            invocationOperation.Syntax.GetLocation(),
+                                            invocationOperation.TargetMethod.ToDisplayString(
+                                                SymbolDisplayFormat.MinimallyQualifiedFormat)));
```

Comment:
[minor] You could use the following syntax:
```suggestion
                                    operationAnalysisContext.ReportDiagnostic(
                                        invocationOperation.CreateDiagnostic(
                                            chosenDiagnostic,
                                            invocationOperation.TargetMethod.ToDisplayString(
                                                SymbolDisplayFormat.MinimallyQualifiedFormat)));
```